src/lib/prov/tpm2/tpm2_algo_mappings.h

```diff
@@ -191,6 +192,50 @@ namespace Botan::TPM2 {
    }
 }
 
+[[nodiscard]] inline std::optional<std::string> curve_id_tss2_to_botan(TPMI_ECC_CURVE mode_id) {
+   // Currently, tpm2-tss does not include support for Brainpool curves or 25519/448.
+   // Once the corresponding PR (https://github.com/tpm2-software/tpm2-tss/pull/2897) is meged and released,
+   // this function should be updated.
+   switch(mode_id) {
+      case TPM2_ECC_NIST_P192:
+         return "secp192r1";
+      case TPM2_ECC_NIST_P224:
+         return "secp224r1";
+      case TPM2_ECC_NIST_P256:
+         return "secp256r1";
+      case TPM2_ECC_NIST_P384:
+         return "secp384r1";
+      case TPM2_ECC_NIST_P521:
+         return "secp521r1";
+      case TPM2_ECC_SM2_P256:
+         return "sm2p256v1";
+      default:
+         return std::nullopt;
+   }
+}
+
+[[nodiscard]] inline std::optional<TPM2_ECC_CURVE> get_tpm2_curve_id(const OID& curve_oid) {
+   // Currently, tpm2-tss does not include support for Brainpool curves or 25519/448.
+   // Once the corresponding PR (https://github.com/tpm2-software/tpm2-tss/pull/2897) is meged and released,
+   // this function should be updated.
+   std::string curve_name = curve_oid.to_formatted_string();
```

Comment:
```suggestion
   const std::string curve_name = curve_oid.to_formatted_string();
```

... this is fairly relevant here. Imagine you introduce a typo and do `if(curve_name = "secp192r1") {`... 😨